docs/essentials/controls.md

```diff
@@ -0,0 +1,208 @@
+
+<div style="background-color:#F8FAFC">
+ask tom/dom/michael for the link of the migration guide.
+</div>
+
+> If you have written stories in the older pre-Storybook 6 style, you may want to read the [args&controls migration guide] to help understand how to convert your stories for args.
+
+### Choosing the control type
+
+By default, Storybook will choose a control for each arg based on the initial value of the arg. This works well with some kind of args, such as boolean values or free-text strings, but in other cases you want a more restricted control.
+
+For instance, suppose you have a `backgroundColor` arg on your story:
+
+```js
+// Button.stories.js
+
+const Red = ButtonStory.bind({});
+Red.args = {
+   backgroundColor: '#e00',
+};
+```
+
+By default, Storybook will render a free text input for the `backgroundColor` arg:
+
+<div style="background-color:#F8FAFC">
+TODO add same image used in the SB 6.0 docs with highlighted background control
+</div>
+
+This works as long as you type a valid string into the auto-generated text control, but it's not the best UI for picking a color. Let’s replace it with Storybook’s color picker component.
+
+We can specify which controls get used by declaring a custom [argType](../api/stories#argtypes) for the `backgroundColor` property. ArgTypes encode basic metadata for args, such as name, description, defaultValue for an arg. These get automatically filled in by Storybook Docs.
+
+ArgTypes can also contain arbitrary annotations which can be overridden by the user. Since `backgroundColor` is a property of the component, let's put that annotation on the default export.
+
+```js
+import { Button } from './Button';
+export default {
+  title: 'Button',
+  component: Button,
+  argTypes: {
+    backgroundColor: { control: 'color' },
+  },
+};
+```
+
+This replaces the input with a color picker leading to a more intuitive developer experience.
+
+<div style="background-color:#F8FAFC">
+TODO add same image used in the SB 6.0 docs color picker opened
+</div>
+
+### Fully custom args
+
+<div style="background-color:#F8FAFC">
+ask tom for location of complex stories link
+</div>
+
+Up until now, we've only used auto-generated controls based on the component we're writing stories for. If we are writing [complex stories](locate-complex-stories) we may want to add controls for args that aren’t part of the component.
+
+```js
+// Table.stories.js
+
+const TableStory = ({ data, ...args }) => (
+  <Table {...args} >
+    {data.map(row => (<TR>{row.map(item => <TD>{item}</TD>}</TR>))}
+  </Table>
+)
+
+export const Numeric = TableStory.bind({});
+Numeric.args = {
+  // This arg is for the story component
+  data: [[1, 2, 3], [4, 5, 6]],
+  // The remaining args get passed to the `Table` component
+  size: 'large',
+}
+```
+
+By default, Storybook will add controls for all args that:
+
+<div style="background-color:#F8FAFC">
+TODO: check if the link to the framework support is correct.
```

Comment:
@shilman?

Reply:
https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#framework-support

Currently: React, Vue, Angular, Ember (WIP), Web components

Reply:
@shilman thanks for the link. Is added. Marking this as resolved.